Sources/GryphonLib/KotlinTranslator.swift

```diff
@@ -730,16 +741,43 @@ public class KotlinTranslator {
 		else {
 			result.append(try translateSubtrees(
 				statements,
-				withIndentation: indentation,
+				withIndentation: !singleExpressionStatement ? indentation : "",
 				limitForAddingNewlines: 3))
 		}
 
 		indentation = decreaseIndentation(indentation)
-		result.append(indentation + "}\n")
+		if !singleExpressionStatement {
+			result.append(indentation + "}\n")
+		}
 
 		return result
 	}
 
+	private func isSingleExpressionFunction(
+		_ statements: List<Statement>?,
+		returnTypeString: String?)
+		-> Bool
+	{
+		guard let statements = statements,
+			let returnTypeString = returnTypeString else {
+				return false
+		}
+
+		guard statements.count == 1,
+			let statement = statements.first,
+			statement is ExpressionStatement else {
+				return false
+		}
+
+		let subtrees = statement.printableSubtrees.compactMap { $0 }
```

Comment:
I don't think you need `printableSubtrees` here, they're mainly for output. You could do something like `guard let expressionStatement = statement as? ExpressionStatement` above, and then just use `expressionStatement.expression` below.

Reply:
Yes, you are absolutely right. 👍